server/routerlicious/packages/lambdas/src/nexus/disconnect.ts

```diff
@@ -116,10 +122,59 @@ export async function disconnectDocument(
 				error,
 			);
 		}
+		// Update session tracker upon disconnection
+		if (collaborationSessionTracker) {
+			const client = clientMap.get(clientId);
+			const connectionTimestamp = connectionTimeMap.get(clientId);
+			if (client) {
+				collaborationSessionTracker.endClientSession(
+					{
+						clientId,
+						joinedTime: connectionTimestamp ?? 0,
+						isSummarizerClient: isSummarizer(client.details),
+						isWriteClient: isWriter(client.scopes, client.mode),
+					},
+					{
+						tenantId: room.tenantId,
+						documentId: room.documentId,
+					},
+				);
+			}
+		}
 	}
+	return promises;
+}
+
+/**
+ * Perform necessary cleanup when a client disconnects from a document.
+ * @internal
```

Comment:
We aren't performing any API trimming on server packages, so this `@internal` tag won't have any effect on the API. Probably best to remove it.